snapcraft/internal/project_loader/_extensions/ros1_noetic.py

```diff
@@ -0,0 +1,83 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2020 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Import types and tell flake8 to ignore the "unused" List.
+
+from typing import Any, Dict, Optional, Tuple
+
+from typing_extensions import Final
+
+from ._extension import Extension
+
+
+class ExtensionImpl(Extension):
+    """Drives ROS 1 build and runtime environment for snap."""
```

Comment:
Cosmetic to follow https://www.python.org/dev/peps/pep-0257/

```suggestion
    """Setup a ROS 1 build and runtime environment suitable for snaps."""
```

This part in the text
> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

You can use my suggestion or write a new string based on that

Reply:
Fixed.